File: C/QuickSort.c
#include <stdio.h>
#include <stdlib.h>

void swap(int *a, int *b)
{
    int t = *a; //temporary
    *a = *b;
    *b = t;
}

int partition(int arr[], int low, int high)
{
    int pivot = arr[high];
    int i = (low - 1);
    int j;
    for ( j = low; j < high; j++)
    {
        if(arr[j] <= pivot)
        {
            i++;
            swap(&arr[i], &arr[j]);
        }
    }
    swap(&arr[i+1], &arr[high]);
    return (i+1);
}

void QuickSort(int arr[], int low, int high)
{
    if(low < high)
    {
        int pi = partition(arr, low, high);
        QuickSort(arr,low, pi -1);
        QuickSort(arr,pi + 1, high);
    }
}

void PrintArray(int arr[], int size)
{
    for(int i = 0; i < size; i++)
    {
        printf("%d ", arr[i]);
    }
}

int main()
{
    int data[] = {4, 6, 3, 17, 10, 15, 12};
    int n = sizeof(data)/sizeof(data[0]);
    printf("Unsorted Array: \n");
    PrintArray(data, n);
    QuickSort(data, 0, n-1);
    printf("\n Sorted Array: \n");
    PrintArray(data, n);
}

File: Python/QuickSort.py
def quick_sort(array):
    if len(array) < 2:
        return array
    else:
        pivot = array[0]
        less = [i for i in array[1:] if i <= pivot]
        greater = [i for i in array[1:] if i > pivot]
        return quick_sort(less) + [pivot] + quick_sort(greater)

arr = [7, 2, 4, 0, 1, 5, 8, 3, 2, 0]
print(quick_sort(arr))